classes/Context.php

```diff
@@ -364,16 +365,23 @@ public function updateCustomer(Customer $customer)
     /**
      * @return Translator
      */
-    public function getTranslator()
+    public function getTranslator($needComponentTranslator = false)
     {
         if (null !== $this->translator) {
             return $this->translator;
         }
 
-        $translator = $this->getTranslatorFromLocale($this->language->locale);
-        $this->translator = $translator;
+        $sfContainer = SymfonyContainer::getInstance();
 
-        return $translator;
+        if ($needComponentTranslator || null === $sfContainer) {
+            // symfony's container isn't available in front office, so we load and configure the translator component
+            $this->translator = $this->getTranslatorFromLocale($this->language->locale);
+        } else {
+            $this->translator = $sfContainer->get('translator');
+            $this->translator->setLocale($this->language->locale);
```

Comment:
On legacy back office pages, the translator is called before listeners do their job of changing the locale, for example there are translations done during AdminController's instantiation.

Reply:
It's better to add this comment in the code.

Reply:
done !  (I did the parameter renaming and phpdoc improvement too, I just can't answer to the comment directly)